turn_on_led/turn_on_led/turn_on_led.asm

```diff
@@ -0,0 +1,659 @@
+	out porta, r16
+	
+	mov r16, blockRow1
+	out portc, r16
+	ret
+
+row_two:
+	ldi r16, 0b00000010
+	out porta, r16
+	
+	mov r16, blockRow2
+	out portc, r16
+	ret
+
+row_three:
+	ldi r16, 0b00000100
+	out porta, r16
+	
+	mov r16, blockRow3
+	out portc, r16
+	ret
+
+row_four:
+	ldi r16, 0b00001000
+	out porta, r16
+	
+	mov r16, blockRow4
+	out portc, r16
+	ret
+
+ball:
+	out porta, ballRow
+	
+	out portc, ballCol
+	ret
+
+paddle:
+	ldi r16, 0b10000000
+	out porta, r16
+	
+	mov r16, paddleCol
+	out portc, r16
+	ret
+
+;delay loop
+delay_10ms:
+	ldi r20, 2
+delay_loop_1:
+    ldi r21, 50 
+delay_loop_2: 
+	ldi r22, 100
+delay_loop_3: ;
+	dec r22
+	nop
+	brne delay_loop_3
+
+	dec r21
+	brne delay_loop_2
+
+	dec r20
+	brne delay_loop_1
+    ret
+
+check_game_over:
+	cpi ballRow, 0b10000000
+	breq sad_face
+	ret
+
+sad_face: ;turn on leds to show a sad face
+	ldi r16, 0b00000001 ;set up first row
+	out porta, r16
+	
+	ldi r16, 0b11111111
+	out portc, r16
+	
+	call delay_10ms
+
+	ldi r16, 0b00000010 ;set up second row
+	out porta, r16
+	
+	ldi r16, 0b10011001
+	out portc, r16
+	
+	call delay_10ms
+
+	
+	ldi r16, 0b00000100 ;set up third row
+	out porta, r16
+	
+	ldi r16, 0b10011001
+	out portc, r16
+	
+	call delay_10ms
+
+	
+	ldi r16, 0b00001000 ;set up fourth row
+	out porta, r16
+	
+	ldi r16, 0b11111111
+	out portc, r16
+	
+	call delay_10ms
+
+	
+	ldi r16, 0b00010000 ;set up fifth row
+	out porta, r16
+	
+	ldi r16, 0b11100111
+	out portc, r16
+	
+	call delay_10ms
+
+	
+	ldi r16, 0b00100000 ;set up sixth row
+	out porta, r16
+	
+	ldi r16, 0b11011011
+	out portc, r16
+	
+	call delay_10ms
+
+	
+	ldi r16, 0b01000000 ;set up seventh row
+	out porta, r16
+	
+	ldi r16, 0b10111101
+	out portc, r16
+	
+	call delay_10ms
+
+	
+	ldi r16, 0b10000000 ;set up eighth row
+	out porta, r16
+	
+	ldi r16, 0b11111111
+	out portc, r16
+	
+	call delay_10ms
+
+	rjmp sad_face
+
+
+start_position: ; turn on leds to show start position
+	;set up first row
+	ldi r16, 0b00000001
+	out porta, r16
+	
+	ldi r16, 0b10101010
+	out portc, r16
+	
+	call delay_10ms
+
+	;set up second row
+	ldi r16, 0b00000010
+	out porta, r16
+	
+	ldi r16, 0b01010101
+	out portc, r16
+	
+	call delay_10ms
+
+	;set up third row
+	ldi r16, 0b00000100
+	out porta, r16
+	
+	ldi r16, 0b10101010
+	out portc, r16
+	
+	call delay_10ms
+
+	;set up fourth row
+	ldi r16, 0b00001000
+	out porta, r16
+	
+	ldi r16, 0b01010101
+	out portc, r16
+	
+	call delay_10ms
+	;set up fifth row
+	ldi r16, 0b00010000
+	out porta, r16
+	
+	ldi r16, 0b11111111
+	out portc, r16
+	
+	call delay_10ms
+
+	;set up sixth row
+	ldi r16, 0b00100000
+	out porta, r16
+	
+	ldi r16, 0b11111111
+	out portc, r16
+	
+	call delay_10ms
+
+	;set up seventh row
+	ldi r16, 0b01000000
+	out porta, r16
+	
+	ldi r16, 0b11101111
+	out portc, r16
+	
+	call delay_10ms
+
+	;set up eighth row
+	ldi r16, 0b10000000
+	out porta, r16
+	
+	ldi r16, 0b11000111
+	out portc, r16
+	
+	call delay_10ms
+
+	ret
+
+
+happy_face: ;turn on leds to show happy face
```

Comment:
Looking at a subroutine like this makes me wonder if there are ways to simplify it. It looks like a there is a lot of almost duplicate code. An alternative approach could have been to store the data you want to display in memory in an array like structure and then simply displayed the data one line at a time until the entire screen was updated. Then you might have been able to make on generic display subroutine.